src/core/thread/discover_scanner.hpp

```diff
@@ -163,6 +181,10 @@ class DiscoverScanner : public InstanceLocator
     uint8_t          mScanChannel;
     bool             mEnableFiltering : 1;
     bool             mShouldRestorePanId : 1;
+    uint32_t         mOui;
+    uint8_t          mAdvData[kMaxLength];
+    uint8_t          mAdvDataLength;
+    bool             mHasJoinerAdvertisement : 1;
```

Comment:
nit: use better data alignment?

- Move bool flags together ?
- Move uint32_t mOui up ?

Reply:
thanks, tried my best to make them look tidy again. :)

Reply:
can mAdvDataLength serve the purpose of mHasJoinerAdvertisement?